modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/UserSetting.kt

```diff
@@ -58,31 +54,26 @@ abstract class UserSetting<T>(
 
     protected abstract fun persist(value: T, commit: Boolean)
 
-    fun set(value: T, commit: Boolean = false, needsSync: Boolean = false) {
+    open fun set(value: T, commit: Boolean = false, needsSync: Boolean = false) {
         persist(value, commit)
         _flow.value = value
-
-        if (needsSync) {
-            updateModifiedAtServerString()
-        }
+        val modifiedAt = if (needsSync) Instant.now().toString() else null
```

Comment:
It would be nice if `UserSettings` somehow accepted `Clock` so we can control the time to make this part more testable. Not a priority for this PR but something to keep in mind.